### DISCUSSION TITLE
Determining excess PV power

### DISCUSSION THREAD

**Discussion by ProtoxiDe22** [nathanmarlor/foxess_modbus#355] [Q&A]
I'm not sure if i'm missing something or this is not possible, but i'm trying to find a way to determine what excess PV power there's available. Obviously the PV power sensor caps at the actual load, so if batteries are charged, there's a good chance that you have more PV power available that you are using, but that isn't shown in the sensor.
On the forum i have read that some other inverter such as the solaredge expose a "Power Limit" register that is a factor that determies how much PV power isn't being used, but if i understand it correctly, this integration (and the foxess protocol spec) doesn't have anything like this
So do we have a way to determine what's the Excess soalr power? if not how do you handle this problem? do you try to optimize your solar consumption at all?

i shall specify that my inverter is an H1

**Comment by nathanmarlor** [comment]
If the power isn't consumed by the house load it's exported to the grid, so I'd assume the inverter is always delivering maximum output - this wouldn't apply in a clipping scenario where PV > inverter output but I can't see a way the inverter would be able to detect that...?

**Reply by ProtoxiDe22** [reply]
In my case, i'm not exporting to grid because of problems with the power company, they have to do works on the infrastructure to handle the output of an inverter of that size at my location. so best case scenario i won't be able to do this until jan 2024 i'd say, since i know i'm not the only one with this problem (i read about people who can't export to grid at all), i hoped there would be an alternate solution

**Comment by canton7** [comment]
Assuming you don't have a battery, and seeing as you can't export to the grid, surely if you're not importing from the grid then either the PV is exactly matched to the load (unlikely), or you're clipping?

If you do have a battery, then similar but if the battery is charging at full rate you're probably not clipping?

**Reply by ProtoxiDe22** [reply]
You make a great point there. this does slightly complicate things since i need to ckeck grid import and battery recharge at the same time but it should be possible nontheless. I'm not the best at complex automations but i'll give it a shot, thank you

**Reply by canton7** [reply]
I'd make a template sensor (or you can use templates as conditions in automations actually I think?)
